### DISCUSSION TITLE
How to train PADIM in detection mode with tiling and custom backbone

### DISCUSSION THREAD

**Discussion by SebastienThibert** [open-edge-platform/anomalib#1139] [Q&A]
Hello,

I'm trying to use PADIM on a custom dataset. I got some results  but I'm not sure if all settings are good:
```
`BACKBONE = 'resnet18' #"resnet18" , 'wide_resnet50_2' et resnet152 to test 
INPUT_SIZE = (256,256)
BATCH_SIZE = 16 # 16 ok avec resnet18 
MAX_EPOCHS = 1 # 1  
TASK = TaskType.DETECTION # TaskType.SEGMENTATION`
```

```
`datamodule = Folder(
    root=RAW_DATA_DIR,
    normal_dir="normal",
    abnormal_dir="abnormal",
    mask_dir = "masks",
    normal_split_ratio=0.2,
    image_size=INPUT_SIZE, # 256,256
    train_batch_size=BATCH_SIZE, 
    eval_batch_size=BATCH_SIZE, 
    task=TASK,
    num_workers=4 
)
    
datamodule.setup()  # Split the data to train/val/test/prediction sets.
datamodule.prepare_data()  # Create train/val/test/predic dataloaders

model = Padim(
    input_size=INPUT_SIZE,
    backbone=BACKBONE,
    layers=["layer1", "layer2", "layer3"],
)

# ORIGINAL 
callbacks = [
    MetricsConfigurationCallback(
        task=TASK,
        image_metrics=["AUROC"], # ["AUROC"]
    ),
    ModelCheckpoint(
        mode="max",
        monitor="image_AUROC", # "image_AUROC"
    ),
    PostProcessingConfigurationCallback(
        normalization_method=NormalizationMethod.MIN_MAX,
        threshold_method=ThresholdMethod.ADAPTIVE,
    ),
    MinMaxNormalizationCallback(),
    ExportCallback(
        input_size=INPUT_SIZE,
        dirpath=str(Path.cwd()),
        filename="model",
        export_mode=ExportMode.OPENVINO,
    ),
    ImageVisualizerCallback(mode="full", task=TASK, image_save_path=TRAINING_IMAGES_OUT_DIR),
    MetricVisualizerCallback(mode="full", task=TASK, image_save_path=TRAINING_IMAGES_OUT_DIR),
]

# training
trainer = Trainer(
    callbacks=callbacks,
    accelerator="auto",
    auto_scale_batch_size=False,
    check_val_every_n_epoch=None,  # 1 
    devices=1,
    max_epochs=MAX_EPOCHS,
    num_sanity_val_steps=0,
    val_check_interval=10, #  # 1.0 
    log_every_n_steps = 1,
)
trainer.fit(model=model, datamodule=datamodule)'


```
My main questions regards: 

- The metric to use during "training" for PADIM ? and the definition of the callbacks ? 
- How to add tiling in a notebook environment ? 
- How to use resnet18 with fine-tuned weights other than ImageNet ? 

Finally I got OoM issue as soon as I moved to wide_resnet50_2 (even with batchsize =1), any best practice ?
